### PR TITLE
Add app_engine.Signer

### LINKED ISSUE #49: Add GAE Signer

### PR DESCRIPTION
**Pull request by jonparrott**
Resolves #49

### CODE REVIEW

File: google/auth/app_engine.py [hunk 1]
```diff
@@ -33,6 +33,31 @@
...
+    def sign(self, message):
+        """Signs a message.
+
+        Args:
+            message (Union[str, bytes]): The message to be signed.
+
+        Returns:
+            bytes: The signature of the message.
+        """
+        # pylint: disable=no-self-use
```
Comment (dhermes) [inline google/auth/app_engine.py]: :-1: on `no-self-use`, just make it a `@staticmethod`

File: google/auth/app_engine.py [hunk 2]
```diff
@@ -33,6 +33,31 @@
...
+        # Locally disabled because this needs to confirm the to Signer
```
Comment (dhermes) [inline google/auth/app_engine.py]: "confirm the to" --> "conform to the"?

File: google/auth/app_engine.py (hunk 1 above)
Comment (jonparrott) [inline google/auth/app_engine.py]: I'm on board with that, but doesn't it break the interface?

Comment (dhermes) [inline google/auth/app_engine.py]: I very much doubt it, but I'm open to learning.

File: google/auth/app_engine.py (hunk 2 above)
Comment (jonparrott) [inline google/auth/app_engine.py]: Removed.

File: google/auth/app_engine.py (hunk 1 above)
Comment (jonparrott) [inline google/auth/app_engine.py]: Switched to staticmethod

**Pull request merged**